_projects/coala_language_server.md

```diff
@@ -0,0 +1,63 @@
+---
+collaborating_projects:
+  - coala
+desc: "Offer coala languager server."
+difficulty: medium
+initiatives:
+  - GSoC
+  - RGSoC
+issues:
+  - "https://github.com/coala/coala-vs-code/issues/3"
+  - "https://github.com/coala/coala-vs-code/issues/4"
+markdown: coala_language_server.md
+mentors:
+  - gaocegege
+name: "coala Language Server"
+requirements:
+  - "The applicant should have at least one patch accepted to any of the coala."
+  - "The applicant shoule be familiar with language server protocol"
+tags:
+  - Language Server
+  - Plugins
+---
+
+The [Language Server Protocol](https://langserver.org/) was created by Microsoft to define a common language for programming language analyzers to speak. Today, several companies have come together to support its growth, including Codenvy, Red Hat, and Sourcegraph, and the protocol is becoming supported by a rapidly growing list of editor and language communities:
+
+- VS Code
+- Eclipse Che
+- Eclipse IDE(Eclipse LSP4E)
+- IntelliJ / JetBrains IDEs
+- Vim
+- Atom (WIP)
+- Emacs (WIP)
+- Sublime (WIP)
+
+We have a state-of-art implementation in [coala-vs-code](https://github.com/coala/coala-vs-code) but it is designed for VS Code only. We need to design and implement a general mechanism for multiple editors/IDEs support.
+
+The project is divided into two parts:
+
+1. Offer a basic languager server.
+2. Implement a plugin based on the language server.
+
+#### Milestones
+
+##### PREPARATION/BONDING
+
+- An official repository exists with CI.
```

Comment:
why not continue using the `coala-vs-code` repo? if so, there is nothing to be done for this item and it should be removed

Reply:
coala-vs-code is not the place for language server, I think.

Reply:
Could you explain why?
I think I am missing something.

If we have a proper coala LS , why do we need coala-vs-code ?

Reply:
I think we have consensus after a gitter chat, that if this project is selected for gsoc, coala admins will create a clone of `coala-vs-code` , and will archive the `coala-vs-code` repo, and the gsoc project will occur in a different repository.  We will then re-assess the viability/utility of two repos after the project is completed.